src/docgen/view/documentation.ts

```diff
@@ -1,121 +1,227 @@
+    return withTempDir(async (workdir: string) => {
+
+      // always better not to operate on an externally provided directory
+      await fs.copy(assembliesDir, workdir);
+
+      let transpile, language;
+
+      switch (options?.language ?? 'ts') {
+        case 'python':
+          language = TargetLanguage.PYTHON;
+          transpile = new PythonTranspile();
+          break;
+        case 'ts':
+          transpile = new TypeScriptTranspile();
+          break;
+        default:
+          throw new Error(`Unsupported language: ${options?.language}`);
+      }
+      const assembly = await createAssembly(assemblyName, workdir, options?.loose ?? true, language);
+      return new Documentation(assembly, transpile);
+    });
+
+  }
+
+  private constructor(
+    private readonly assembly: reflect.Assembly,
+    private readonly transpile: Transpile) {
   }
 
   /**
    * Generate markdown.
    */
-  public render(): Markdown {
+  public render(options?: RenderOptions): Markdown {
+    const submodule = options?.submodule ? this.findSubmodule(this.assembly, options.submodule) : undefined;
     const documentation = new Markdown();
 
-    if (this.includeReadme) {
-      const readme = new Readme(this.transpile, this.assembly, this.submodule);
+    if (options?.readme ?? true) {
+      const readme = new Readme(this.transpile, this.assembly, submodule);
       documentation.section(readme.render());
     }
-    if (this.includeApiReference) {
-      const apiReference = new ApiReference(
-        this.transpile,
-        this.assembly,
-        this.submodule,
-      );
+
+    if (options?.apiReference ?? true) {
+      const apiReference = new ApiReference(this.transpile, this.assembly, submodule);
       documentation.section(apiReference.render());
     }
 
     return documentation;
   }
 
   /**
-   * Lookup a submodule by a submodule name given by the caller.
+   * Lookup a submodule by a submodule name.
    */
-  private findSubmodule(submoduleName: string): reflect.Submodule {
-    const submodules = this.assembly.submodules.filter(
-      (s) => s.name === submoduleName,
+  private findSubmodule(assembly: reflect.Assembly, submodule?: string): reflect.Submodule {
+    const submodules = assembly.submodules.filter(
+      (s) => s.name === submodule,
     );
 
     if (submodules.length === 0) {
-      throw new Error(`Submodule ${submoduleName} not found`);
+      throw new Error(`Submodule ${submodule} not found`);
     }
 
     if (submodules.length > 1) {
-      throw new Error(`Found multiple submodules with name: ${submoduleName}`);
+      throw new Error(`Found multiple submodules with name: ${submodule}`);
     }
 
     return submodules[0];
   }
 }
+
+async function withTempDir<T>(work: (workdir: string) => Promise<T>): Promise<T> {
+  const workdir = await fs.mkdtemp(path.join(os.tmpdir(), path.sep));
```

Comment:
```suggestion
  const workdir = await fs.mkdtemp(path.join(os.tmpdir(), ''));
```

Reply:
Why? This will actually break because it will produce directories like `/tmp123456`, as opposed to `/tmp/123456.

Reply:
I usually do: `await fs.mkdtemp(path.join(os.tmpdir(), 'jsii-docgen.'));`